python/S3_PUBLIC_ACCESS_SETTINGS_FOR_ACCOUNT/S3_PUBLIC_ACCESS_SETTINGS_FOR_ACCOUNT.PY

```diff
@@ -0,0 +1,439 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  S3_PUBLIC_ACCESS_SETTINGS_FOR_ACCOUNT
+Description:
+  Checks whether AWS S3 public access account settings match the assigned parameters.
+Trigger:
+  Periodic
+Resource Type to report on:
+  AWS::::Account
+Rule Parameters:
+  | ---------------------- | ---------- | -----------------------------------------------------------------------------------------|
+  | Parameter Name         | Type       | Description                                                                              |
+  | ---------------------- | ---------- | -----------------------------------------------------------------------------------------|
+  | BlockPublicAcls        | Optional   | Block new public ACLs and uploading public objects (Default True)                        |
+  | IgnorePublicAcls       | Optional   | Remove public access granted through public ACLs (Default True)                          |
+  | BlockPublicPolicy      | Optional   | Block new public bucket policies (Default True)                                          |
+  | RestrictPublicBuckets  | Optional   | Block public and cross-account access to buckets that have public policies (Default True)|
+  | ---------------------- | ---------- | -----------------------------------------------------------------------------------------|
+Feature:
+  In order to: ensure that s3 account settings are being restricted to the appropriate level
+           As: a Security Officer
+       I want: to verify that the configuration of s3 account public access settings is correct.
+Scenarios:
+  Scenario 1:

```

Comment:
Need a first scenario on if one or more input parameters are not valid.

Reply:
Added new scenario 1 which states that if input parameters are invalid an error is thrown.